### PR TITLE
type-strings-into-constants Extracted blurr types into string constants.

### PR DESCRIPTION
**Pull request by vigbk**
**PR Checklist**

_Please make sure to review and check all of these items:_

- [x] Tests for the changes have been added
- [x] Docs have been added / updated

**What kind of change does this PR introduce?**

_Describe the big picture of your changes here_

**Other information**

fixes #XYZ

### CODE REVIEW

**Comment by vigbk** [comment]
I think we should have some consistent naming in our types. Sometimes we append the type to the end sometimes we don't. eg: Blurr:Transform:Streaming, Blurr:Aggregate:VariableAggregate. What do you guys think?

File: blurr/core/constants.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+BLURR_TRANSFORM_STREAMING = "Blurr:Transform:Streaming"
+BLURR_TRANSFORM_WINDOW = "Blurr:Transform:Window"
+BLURR_AGGREGATE_BLOCK = "Blurr:Aggregate:BlockAggregate"
```
Comment (indradeep) [inline blurr/core/constants.py]: Let us include the change we discussed here - for Aggregates, remove the superfluous `Aggregate` in the end.  Remember to do a mass replace form the documentation too!

File: blurr/core/constants.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+BLURR_TRANSFORM_STREAMING = "Blurr:Transform:Streaming"
```
Comment (prashastk) [inline blurr/core/constants.py]: How about creating a type.py file which defines these as an enum:
```
class Type(Enum):
    BLURR_TRANSFORM_STREAMING = "Blurr:Transform:Streaming",
    BLURR_AGGREGATE_BLOCK = "Blurr:Aggregate:BlockAggregate",
```

Then passing around Types will be clear as an object as opposed to arbitrary strings. This class can also encapsulate other type related functionality that we may want to add in the future.

File: blurr/core/constants.py (hunk 1 above)
Comment (vigbk) [reply blurr/core/constants.py]: Will do

File: blurr/core/constants.py (hunk 2 above)
Comment (vigbk) [reply blurr/core/constants.py]: Sure. Probably Call it DTCType to be more specific.

Comment (prashastk) [reply blurr/core/constants.py]: file being in blurr/core makes it about the DTC. don't think we need to specify that (just like we don't call other classes / objects as DTC)

**Comment by vigbk** [comment]
Okay sure

File: blurr/core/constants.py (hunk 2 above)
Comment (vigbk) [reply blurr/core/constants.py]: sure

**Comment by vigbk** [comment]
Updated PR after converting from string constants to enums.
Also removed redundant suffixes from string types eg: Blurr:Aggregate:VariableAggregate -> Blurr:Aggregate:Variable.

I will update the documentation in another commit for ease of review.

**Pull request merged**